### ISSUE TITLE
When will Windows Authentication for Blazor Web App template be added to Visual Studio 2022 Preview 3

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/aspnetcore#51860]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/When-will-Windows-Authentication-for-Bla/10491928)._

---
When will "Windows Authentication" be added to the "Authentication type" list in the Blazor Web App template for Visual Studio 2022 Preview 3, please?
Thanks

---
#### Feedback Bot on 10/15/2023, 06:53 PM: 

(private comment, text removed)

**Comment by javiercn** [comment]
# Enabling Windows Authentication in an Existing Blazor App

## 1️⃣ Program.cs

Add Negotiate authentication, authorization, and authentication state support.

```csharp
using Microsoft.AspNetCore.Authentication.Negotiate;

var builder = WebApplication.CreateBuilder(args);

// Add Razor Components (Blazor Server / Interactive Server)
builder.Services.AddRazorComponents()
    .AddInteractiveServerComponents();

// Add Windows Authentication
builder.Services.AddAuthentication(NegotiateDefaults.AuthenticationScheme)
    .AddNegotiate();

// Require authentication by default
builder.Services.AddAuthorization(options =>
{
    options.FallbackPolicy = options.DefaultPolicy;
});

// Make authentication state available to components
builder.Services.AddCascadingAuthenticationState();

var app = builder.Build();

app.UseHttpsRedirection();
app.UseStaticFiles();
app.UseRouting();

app.UseAuthentication();
app.UseAuthorization();

app.MapRazorComponents<App>()
    .AddInteractiveServerRenderMode();

app.Run();
```

## 2️⃣ Razor Component Example

Protect a page with `[Authorize]`:

```razor
@page "/secure"
@attribute [Authorize]

<h3>Secure Page</h3>
<p>Welcome, @context.User.Identity.Name</p>
```

Show content conditionally with `<AuthorizeView>`:

```razor
<AuthorizeView>
    <Authorized>
        Hello, @context.User.Identity.Name!
    </Authorized>
    <NotAuthorized>
        You are not authorized to view this content.
    </NotAuthorized>
</AuthorizeView>
```

## 3️⃣ launchSettings.json

Located in `Properties/launchSettings.json`.

```json
{
  "iisSettings": {
    "windowsAuthentication": true,
    "anonymousAuthentication": false,
    "iisExpress": {
      "applicationUrl": "http://localhost:5000",
      "sslPort": 44300
    }
  },
  "profiles": {
    "IIS Express": {
      "commandName": "IISExpress",
      "launchBrowser": true,
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    },
    "BlazorApp": {
      "commandName": "Project",
      "dotnetRunMessages": true,
      "applicationUrl": "http://localhost:5000;https://localhost:5001",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    }
  }
}
```

## 4️⃣ Project File (.csproj)

Add these properties to enable Windows Authentication in Visual Studio’s generated settings:

```xml
<PropertyGroup>
  <WindowsAuthentication>true</WindowsAuthentication>
  <AnonymousAuthentication>false</AnonymousAuthentication>
</PropertyGroup>
```
  
## TL;DR

- **Program.cs** → Add Negotiate authentication, authorization, and authentication state.
- **Razor Components** → Use `[Authorize]` and `<AuthorizeView>`.
- **launchSettings.json** → Enable Windows Auth for IIS Express.
- **.csproj** → Set WindowsAuthentication to `true` and AnonymousAuthentication to `false`.

---
